src/library/scala/beans/ScalaBeanInfo.scala

```diff
@@ -32,15 +31,9 @@ abstract class ScalaBeanInfo(clazz: java.lang.Class[_],
   override def getPropertyDescriptors() = pd
   override def getMethodDescriptors() = md
 
-  // override def getAdditionalBeanInfo() = Array(Introspector getBeanInfo clazz.getSuperclass)
-
   private def init() {
-    var i = 0
```

Comment:
Oh how I _wish_ foreach on ranges were inlined and transformed!  This has been proposed over and over, and shot down by Martin for being insufficiently general; implemented in plugins and macros by Olivier Chafik (ScalaCL, Scalaxy) and Paul Phillips and at least one other; and it's still not in.  A simple while loop skipping by 3, even on optimized, turns from a compact local 31-byte method into a hulking 189-byte monstrosity with over a dozen method calls.  Not having to write while loops would save me _so_ much grief.  Anyway, I don't worry about the general case where there are many objects, but about the case where there are very few properties (zero or one), which I suspect is probably quite common.